File: solutions/elixir/line-up/1/lib/line_up.ex
defmodule LineUp do

  def suffix(number) do
    case number do
      n when rem(n, 100) in [11, 12, 13] -> "th"
      n when rem(n, 10) == 1 -> "st"
      n when rem(n, 10) == 2 -> "nd"
      n when rem(n, 10) == 3 -> "rd"
      _ -> "th"
    end
  end

  def format(name, number) do
    "#{name}, you are the " <>
    "#{number}#{suffix(number)} " <>
    "customer we serve today. Thank you!"
  end

end

File: solutions/javascript/line-up/1/line-up.js
export const format = (name, number) => {
  let ordinalSuffix = 'th'

  if (number % 10 === 1 && number % 100 !== 11) {
    ordinalSuffix = 'st'
  } else if (number % 10 === 2 && number % 100 !== 12) {
    ordinalSuffix = 'nd'
  } else if (number % 10 === 3 && number % 100 !== 13) {
    ordinalSuffix = 'rd'
  }

  return `${name}, you are the ${number}${ordinalSuffix} customer we serve today. Thank you!`
}
